cmd/agent/subcommands/run/command.go

```diff
@@ -402,6 +405,39 @@ func getSharedFxOption() fx.Option {
 			}
 		}),
 		healthprobeimpl.Module(),
+		settingsRegistryimpl.Module(),
+		settingsServerimpl.Module(),
+		fx.Provide(func() settingsRegistry.RuntimeSettingProvider {
+			return settingsRegistry.NewRuntimeSettingProvider(commonsettings.NewLogLevelRuntimeSetting())
+		}),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** 
Would it be possible to replace the Provide call with Supply for a more straightforward registration ?‏
```suggestion
fx.Supply(settingsRegistry.NewRuntimeSettingProvider(commonsettings.NewLogLevelRuntimeSetting())),
```